CMakeLists.txt

```diff
@@ -14,6 +14,37 @@ set(INSTALL_CMAKE_DIR lib/cmake CACHE PATH "Installation directory for cmake fil
 
 set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
 
+set(CTEST_MEMORYCHECK_TYPE "Valgrind")
+
+set(MEMORYCHECK_COMMAND_OPTIONS_REGULAR " \
+    --leak-check=full \
+    --leak-resolution=high \
+    --trace-children=yes \
+    --run-libc-freeres=no \
+    -q --error-exitcode=123 \
+    --error-limit=no \
+    --num-callers=40 \
+    --undef-value-errors=no \
+    --suppressions=valgrind.suppressions")
+
+set(MEMORYCHECK_COMMAND_OPTIONS_PENDANTIC " \
+    --leak-check=full \
+    --leak-resolution=high \
+    --trace-children=yes \
+    --run-libc-freeres=no \
+    -q --error-exitcode=123 \
+    --error-limit=no \
+    --num-callers=40 \
+    --undef-value-errors=no \
+    --errors-for-leak-kinds=all \
+    --suppressions=valgrind.suppressions")
+
+if ($ENV{S2N_LIBCRYPTO} MATCHES "openssl-1.1.1")
```

Comment:
More generally, I'd like to include a comment about
- pedantic vs regular valgrind
- why we are only running pedantic valgrind on openssl-1.1.1

Reply:
I have added the comments to indicate that we are doing pedantic Valgrind check for openssl-1.1.1 per tracking issue #3758. We will add pedantic Valgrind check incrementally, so we first do Libcrypto == openssl1.1.1 for now.